src/charm.py

```diff
@@ -345,6 +354,11 @@ def _update_workload(self, event) -> None:
             event.defer()
             return
 
+        # Update the ssh ingress to reflect ssh port config changed. This is done in the leader unit
+        # because the ingress is per-unit and it doesn't support multiple units.
+        if self.unit.is_leader():
+            self.ingress_ssh.provide_ingress_requirements(port=self.config.get("ssh-port"))
```

Comment:
shouldn't we do this in the leader elected hook?
register a hook like
```
self.framework.observe(self.on.leader_elected, self._on_leader_elected)
```
in init and then do this in the _on_leader_elected if unit is the leader..